### PR TITLE
[Storage] Open db in parallel when storage sharding is enabled.

### PR DESCRIPTION
**Pull request by grao1991**
## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [x] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [x] Validator Node
- [x] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: storage/aptosdb/src/state_kv_db.rs [hunk 1]
```diff
@@ -56,19 +57,13 @@ impl StateKvDb {
             });
         }
 
-        Self::open(
-            db_paths,
-            rocksdb_configs.state_kv_db_config,
-            readonly,
-            sharding,
-        )
+        Self::open(db_paths, rocksdb_configs.state_kv_db_config, readonly)
```
Comment (msmouse) [inline storage/aptosdb/src/state_kv_db.rs:60]: let's rename to `Self::open_sharded()` or something?

File: storage/aptosdb/src/ledger_db/mod.rs [hunk 2]
```diff
@@ -155,60 +156,95 @@ impl LedgerDb {
...
-            &rocksdb_configs.ledger_db_config,
-            readonly,
-        )?));
+        let mut event_db = None;
+        let mut transaction_accumulator_db = None;
+        let mut transaction_auxiliary_data_db = None;
+        let mut transaction_db = None;
+        let mut transaction_info_db = None;
+        let mut write_set_db = None;
+        THREAD_MANAGER.get_non_exe_cpu_pool().scope(|s| {
```
Comment (msmouse) [inline storage/aptosdb/src/ledger_db/mod.rs:165]: seems no particular reason to use this pool?

Comment (grao1991) [reply storage/aptosdb/src/ledger_db/mod.rs:165]: No. Just need some threads here.

File: storage/aptosdb/src/state_kv_db.rs (hunk 1 above)
Comment (grao1991) [reply storage/aptosdb/src/state_kv_db.rs]: Done

**Review by msmouse — APPROVED** [review]

File: storage/aptosdb/src/db_options.rs [hunk 3]
```diff
@@ -221,15 +221,8 @@ pub(super) fn gen_state_merkle_cfds(rocksdb_config: &RocksdbConfig) -> Vec<Colum
...
-pub(super) fn gen_state_kv_cfds(
-    rocksdb_config: &RocksdbConfig,
-    enable_sharding: bool,
-) -> Vec<ColumnFamilyDescriptor> {
-    let cfs = if enable_sharding {
-        state_kv_db_new_key_column_families()
-    } else {
-        state_kv_db_column_families()
-    };
+pub(super) fn gen_state_kv_cfds(rocksdb_config: &RocksdbConfig) -> Vec<ColumnFamilyDescriptor> {
```
Comment (msmouse) [inline storage/aptosdb/src/db_options.rs:224]: probably `gen_state_kv_shard_cfds()`

Comment (grao1991) [reply storage/aptosdb/src/db_options.rs]: Done

**Review by areshand — APPROVED** [review]

**Pull request merged**